src/ops/attention.cc

```diff
@@ -269,6 +263,12 @@ MultiHeadAttention::MultiHeadAttention(FFModel &model,
       qoSeqLength(_query->dims[1].size), kvSeqLength(_key->dims[1].size)
 // bias_initializer(_bias_initializer)
 {
+  // assert shape is valid
```

Comment:
Is there a way we could use delegating constructors so this isn't duplicated? If no that's fine, though I'd want a brief explanation why not

Reply:
The input checks being duplicated? I think those can be moved to the `Op` constructor as mentioned below.